### DISCUSSION TITLE
How to set hard size for image and apply scaling?

### DISCUSSION THREAD

**Discussion by MairwunNx** [korlibs/korge#2106] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/39625750/151706095-356ed25b-2647-4f3d-b72b-9ea982c4254d.png)

I tried use this code and it's does not work as I except, maybe exist more correctly way to resize image and apply resize strategy (fit, crop etc).

After this code processing my image looks like just cutted.

**Comment by soywiz** [comment]
How would you expect it to work?
`ScaleMode.COVER` with `Anchor.TOP_LEFT` as put there, it is going to crop either the right part or the bottom part. Usually with COVER you would use `Anchor.CENTER`. The other scale modes are available here: https://github.com/korlibs/korma/blob/4762b2ba981868f0f1c5f82404511413fd109918/korma/src/commonMain/kotlin/com/soywiz/korma/geom/ScaleMode.kt#L29-L51

**Comment by MairwunNx** [comment]
@soywiz I expected something like "crop and fit" when you've large picture and we can use ratio for cropping

**Comment by soywiz** [comment]
@MairwunNx Typical crop & fit should be `ScaleMode.COVER` with `Anchor.CENTER`

**Comment by MairwunNx** [comment]
@soywiz

I tried to use 

```kotlin
 bitmap = BitmapSlice(
     korge_png.get().bmp.resized(450, 450, ScaleMode.FIT, Anchor.MIDDLE_CENTER),
     RectangleInt(0, 0, 450, 450)
 )
```

But it's don't like a crop and fit
![image](https://user-images.githubusercontent.com/39625750/151886073-3fb02a15-a05d-4c99-988e-f081474ced4f.png)

**Comment by soywiz** [comment, marked as answer]
@MairwunNx How is the image you put generated? Are you using the latest 2.4.10 version of korim?

By using the code you provided I get the following results (for a square image fitting in a square region, it acts as just scaling and is the expected behaviour): 
![2022-02-01_0-49-32](https://user-images.githubusercontent.com/570848/151892070-4dd0e89c-0bef-4d9d-9791-24584b422d66.png)

Fitting a square image into a rectangular space just works as expected too:
![2022-02-01_0-50-16](https://user-images.githubusercontent.com/570848/151892072-101e4cc3-3df1-405d-8ae6-387aaf19b374.png)

And `Anchor.COVER` works like this:
![2022-02-01_0-50-45](https://user-images.githubusercontent.com/570848/151892068-3e07e262-f6be-4d22-ab91-fcf74adde328.png)
